src/sllidar_node.cpp

```diff
@@ -138,14 +138,24 @@ class SLlidarNode : public rclcpp::Node
                     return true;
                 case SL_LIDAR_STATUS_ERROR:
                     RCLCPP_ERROR(this->get_logger(),"Error, SLLidar internal error detected. Please reboot the device to retry.");
-                    return false;
+                    return manage_internal_error();
+                    // return false;
             }
         } else {
             RCLCPP_ERROR(this->get_logger(),"Error, cannot retrieve SLLidar health code: %x", op_result);
             return false;
         }
     }
 
+    bool manage_internal_error()
+    {
+        RCLCPP_INFO(this->get_logger(),"Reseting SLLidar");
+        sl_result reset_result;
+        reset_result = drv->reset();
+        RCLCPP_INFO(this->get_logger(),"Reset ok?: %x", reset_result);
```

Comment:
If there is a chance that the reset fails, than add n tries and return false if that happens